### PR TITLE
Add Duration.toMillis() method and tests.

### PR DESCRIPTION
**Pull request by songy23**
This method is needed to calculate interval stats based on time buckets, because buckets may be incomplete and we need to calculate the ratio of passed time / bucket duration. E.g, a time bucket starts at timestamp 0 and has a duration of 20 seconds, and now is 14th seconds, the aggregated data of this bucket should be weighted as `toMillis(14 - 0) / toMillis(20)` = 0.7. 

/cc @dinooliva

### CODE REVIEW

File: api/src/main/java/io/opencensus/common/Duration.java [hunk 1]
```diff
@@ -71,6 +71,15 @@ public static Duration fromMillis(long millis) {
   }
 
   /**
+   * Returns the milliseconds representation of this {@code Duration}.
+   *
+   * @return the milliseconds representation of this {@code Duration}.
+   */
+  public long toMillis() {
```
Comment (bogdandrutu) [inline api/src/main/java/io/opencensus/common/Duration.java]: Why do you need this?

Comment (songy23) [inline api/src/main/java/io/opencensus/common/Duration.java]: It's mainly for computing ratio between two Durations. I've moved this method to implcore instead of the api package.

File: core_impl/src/main/java/io/opencensus/implcore/stats/MutableViewData.java [hunk 2]
```diff
@@ -221,6 +225,16 @@ Timestamp getStart() {
   }
 
   /**
+   * Returns the milliseconds representation of this {@code Duration}.
+   *
+   * @param duration {@code Duration}.
+   * @return the milliseconds representation of a {@code Duration}.
+   */
+  static long toMillis(Duration duration) {
```
Comment (dinooliva) [inline core_impl/src/main/java/io/opencensus/implcore/stats/MutableViewData.java]: This is package-private rather than private for testing, correct? So two questions:
1. In open source, do we annotate with @VisibleForTesting or mark this in any way?
2. Should the comments be javadoc? I seem to recall that we were only going to add javadoc for public things but I don't recall if that changed.

Comment (bogdandrutu) [inline core_impl/src/main/java/io/opencensus/implcore/stats/MutableViewData.java]: We use  @VisibleForTesting if that is the case.

Comment (songy23) [inline core_impl/src/main/java/io/opencensus/implcore/stats/MutableViewData.java]: There will be other classes using this method, so I think it's not strictly `VisibleForTesting`. I agree since this method is package-private, using javadoc instead of plain comments might be unnecessary. I've updated the comments here.

Comment (dinooliva) [inline core_impl/src/main/java/io/opencensus/implcore/stats/MutableViewData.java]: If this method will be used across different classes, is there a utility class that it should be part of?

Comment (songy23) [inline core_impl/src/main/java/io/opencensus/implcore/stats/MutableViewData.java]: Currently we don't have such a util class in implcore. It might make sense to put all the static package-private methods together in a util class, though I think that will be in another PR.

Comment (dinooliva) [reply core_impl/src/main/java/io/opencensus/implcore/stats/MutableViewData.java]: Something like this is clearly a utility that should probably be in a util class - that may not be true for all package private methods but agree, we can look into that in a subsequent pr.
